packages/swapper/src/swappers/osmosis/OsmosisSwapper.ts

```diff
@@ -0,0 +1,333 @@
+import { AssetId, ChainId, cosmosChainId, osmosisChainId } from '@shapeshiftoss/caip'
+import { cosmos, osmosis } from '@shapeshiftoss/chain-adapters'
+import { bip32ToAddressNList } from '@shapeshiftoss/hdwallet-core'
+import { Asset } from '@shapeshiftoss/types'
+import axios from 'axios'
+
+import {
+  ApprovalNeededOutput,
+  BuildTradeInput,
+  BuyAssetBySellIdInput,
+  ExecuteTradeInput,
+  GetTradeQuoteInput,
+  MinMaxOutput,
+  SwapError,
+  SwapErrorTypes,
+  Swapper,
+  SwapperType,
+  Trade,
+  TradeQuote,
+  TradeResult,
+  TradeTxs
+} from '../../api'
+import { bn, bnOrZero } from '../utils/bignumber'
+import { DEFAULT_SOURCE, FEE, GAS, MAX_SWAPPER_SELL } from './utils/constants'
+import {
+  getRateInfo,
+  IsymbolDenomMapping,
+  performIbcTransfer,
+  pollForAtomChannelBalance,
+  pollForComplete,
+  symbolDenomMapping
+} from './utils/helpers'
+import { OsmoSwapperDeps } from './utils/types'
+export class OsmosisSwapper implements Swapper<ChainId> {
```

Comment:
Can you clarify if this is an osmosis specific swapper or intended to be a cosmos-sdk/ibc swapper implementation?

Reply:
This is an Osmosis specific swapper

Reply:
Its a cosmos-sdk/ibc swapper that uses osmosis to trade ibc assets. It currently only supports osmo & atom but could later be refactored to support any number of ibc assets that can interact with osmosis

Reply:
yea, but it uses osmosis specifically to make the transfers

Reply:
Thats definitely not the way I see it.

Currently swapper supports cosmos (an ibc chain) and osmosis (and ibc chain that the swapping happens on)

if we added juno support it should still be in the osmosis swapper.

Reply:
no, that's not what I'm saying.  I'm saying if we wanted to support swapping THROUGH juno, then we would need another swapper